### PR TITLE
BLD: update wheel builds on macos to macos-12 image

### PR DESCRIPTION
**Pull request by mattip**
Redo the part of #22534 that didn't make it into the much larger #22525

xref @charris

### CODE REVIEW

**Comment by charris** [comment]
:1st_place_medal:  Beat me to it.

**Comment by charris** [comment]
close/ropen

**Comment by mattip** [comment]
I am seeing 
```
NFO: /usr/local/bin/gfortran -Wall -g -arch x86_64 -Wall -g -undefined dynamic_lookup -bundle /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpv0zett_w/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpv0zett_w/src.macosx-10.9-x86_64-3.8/_test_abstract_interface_TestAbstractInterface_ext_modulemodule.o /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpv0zett_w/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpv0zett_w/src.macosx-10.9-x86_64-3.8/fortranobject.o /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpv0zett_w/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpt2nb9pct/foo.o /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpv0zett_w/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpv0zett_w/src.macosx-10.9-x86_64-3.8/_test_abstract_interface_TestAbstractInterface_ext_module-f2pywrappers2.o -L/opt/gfortran-darwin-x86_64-native/bin/../lib/gcc/x86_64-apple-darwin13.4.0/11.3.0 -L/opt/gfortran-darwin-x86_64-native/bin/../lib/gcc/x86_64-apple-darwin13.4.0/11.3.0/../../.. -L/opt/gfortran-darwin-x86_64-native/bin/../lib/gcc/x86_64-apple-darwin13.4.0/11.3.0/../../.. -lgfortran -o ./_test_abstract_interface_TestAbstractInterface_ext_module.cpython-38-darwin.so

   ld: library not found for -lm
```

**Comment by charris** [comment]
The f2py tests are failing, same as before.

**Comment by charris** [comment]
Almost there.
```
  + python /Users/runner/work/numpy/numpy/tools/openblas_support.py --check_version
  Traceback (most recent call last):
    File "/Users/runner/work/numpy/numpy/tools/openblas_support.py", line 319, in <module>
      test_version(args.check_version)
    File "/Users/runner/work/numpy/numpy/tools/openblas_support.py", line 297, in test_version
      raise ValueError(f"expected single threadpool_info result, got {data}")
  ValueError: expected single threadpool_info result, got []
```
Wonder what the problem was. I'll be glad to see the end of Python 3.8 in 1.25.x, it just causes us problems.

**Comment by mattip** [comment]
Strange. The delocate command is packaging OpenBLAS into the wheel, and the tests pass so it must be finding some blas to run. But threadpoolctl cannot find any information. The [cpython3.9 run](https://github.com/numpy/numpy/actions/runs/3491514775/jobs/5844241543#step:5:4822) works fine.

**Comment by mattip** [comment]
I am rerunning the job.

**Comment by mattip** [comment]
Grasping at straws. I don't see any threadpoolctl issues around detection on python3.8, and the PyPy3.8 build passes so I don't understand what is wrong with CPython3.8.

**Comment by mattip** [comment]
It fails faster now :slightly_frowning_face: I will try disabling the check just so I can grab the artifact and try it locally

File: azure-pipelines.yml [hunk 1]
```diff
@@ -190,22 +191,16 @@ stages:
...
         CC: /usr/bin/clang
-      condition: eq(variables['USE_OPENBLAS'], '1')
-    - script: python setup.py build -j 4 build_ext --inplace install
-      displayName: 'Build NumPy without OpenBLAS and new casting'
-      env:
-        BLAS: None
-        LAPACK: None
-        ATLAS: None
-        CC: /usr/bin/clang
-      condition: eq(variables['USE_OPENBLAS'], '0')
```
Comment (mattip) [inline azure-pipelines.yml:201]: At one point these were different, but then we removed an env variable so they became the same

File: tools/wheels/cibw_test_command.sh [hunk 2]
```diff
@@ -15,17 +15,18 @@ fi
...
+    export LIBRARY_PATH="$LIBRARY_PATH:/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib"
 fi
 # Set available memory value to avoid OOM problems on aarch64.
 # See gh-22418.
 export NPY_AVAILABLE_MEM="4 GB"
 if [[ $(python -c "import sys; print(sys.implementation.name)") == "pypy" ]]; then
-  # make PyPy more verbose, try to catc a segfault in
-  # numpy/lib/tests/test_function_base.py
+  # make PyPy more verbose, try to catch a segfault
   python -c "import sys; import numpy; sys.exit(not numpy.test(label='full', verbose=2))"
```
Comment (mattip) [inline tools/wheels/cibw_test_command.sh:28]: Could probably remove the verbosity increment, but PyPy is flaky enough that it makes sense to be defensive here.

**Comment by mattip** [comment]
It seems the macos-11 image has some problems, I changed this to use macos-12 instead

**Comment by charris** [comment]
Was going to say ... :)

**Comment by mattip** [comment]
All checks pass. If it is acceptable to skip the test on the macos 3.8 wheel build, this is ready for review.

**Comment by charris** [comment]
> If it is acceptable to skip the test on the macos 3.8 wheel build

Fine with me. We can drop the 3.8 wheel builds after the 1.24 branch.

**Comment by charris** [comment]
@mattip ISTR ILP64 problems for aarch64. Has that been addressed?

**Comment by mattip** [comment]
aarch64 has both 32- and 64-bit interfaces tested. ppc64le was the one with only 32-bit interfaces available

**Comment by mattip** [comment]
xref #22610

**Review by charris — APPROVED** [review]

**Comment by charris** [comment]
Here we go. Thanks Matti.

**Pull request merged**